hail/python/hailtop/aiogoogle/client/storage_client.py

```diff
@@ -644,9 +644,17 @@ async def _listfiles_recursive(self, bucket: str, name: str) -> AsyncIterator[Fi
             items = page.get('items')
             if items is not None:
                 for item in page['items']:
-                    yield GoogleStorageFileListEntry(f'gs://{bucket}/{item["name"]}', item)
-
-    async def _listfiles_flat(self, bucket: str, name: str) -> AsyncIterator[FileListEntry]:
+                    url = f'gs://{bucket}/{item["name"]}'
+                    status = None
+                    if not force_all and url.endswith('/') and item is not None:
```

Comment:
all is a bit general when you're just controlling empty directory files.  What about `skip_empty_directory_files` (or `include_empty_directory_files` if you want the positive sense, although personally I find the sense that doesn't require a `not` to be more readable.)

Reply:
Also, I don't think `item is not None` is needed here.  Item shouldn't be able to be None (let's add an assertion if we're unsure).

Reply:
The reason I don't like `all` is that it isn't clear what files are excluded when you set `include_all_files=False`.